include/drivers/shiftx_drv.h

```diff
@@ -0,0 +1,147 @@
+ * @param red red color value
+ * @param green green color value
+ * @param blue blue color value
+ * @param flash flash rate, in Hz
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_discrete_led(uint8_t led_index, uint8_t leds_to_set, uint8_t red, uint8_t green, uint8_t blue, uint8_t flash);
+
+/**
+ * Set the display value
+ * @param digit_index index of digit to set
+ * @param ascii ascii value to set (48 = '0', 49 = '1')
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_display(uint8_t digit_index, uint8_t ascii);
+
+/**
+ * Configures the linear graph on the ShiftX device
+ * @param rendering_style 0=left->right; 1=center; 2=right->left
+ * @param linear_style 0=smooth/interpolated; 1=stepped
+ * @param low_range low value range of linear display
+ * @param high_range high value range of linear display (ignored if rendering_style = stepped)
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_config_linear_graph(uint8_t rendering_style, uint8_t linear_style, uint16_t low_range, uint16_t high_range);
+
+/**
+ * Configures a threshold for the linear graph on the ShiftX device
+ * @param threshold_id id of threshold (0-4)
+ * @param segment_length number of segments to display for this threshold (ignored if linear style = smooth)
+ * @param threshold value for this threshold
+ * @param red red color value
+ * @param green green color value
+ * @param blue blue color value
+ * @param flash flash rate, in Hz
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_linear_threshold(uint8_t threshold_id, uint8_t segment_length, uint16_t threshold, uint8_t red, uint8_t green, uint8_t blue, uint8_t flash);
+
+/**
+ * Updates the current value for the linear graph
+ * @param value the current runtime value
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_update_linear_graph(uint16_t value);
+
+/**
+ * Configures a threshold for an alert on the ShiftX device
+ * @param alert_id id of alert (0-> number of alert indicators on device)
+ * @param threshold_id id of threshold (0-4)
+ * @param threshold value for this threshold
+ * @param red red color value
+ * @param green green color value
+ * @param blue blue color value
+ * @param flash flash rate, in Hz
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_alert_threshold(uint8_t alert_id, uint8_t threshold_id, uint16_t threshold, uint8_t red, uint8_t green, uint8_t blue, uint8_t flash);
+
+/**
+ * Directly set an alert indicator
+ * @param alert_id id of alert (0-> number of alert indicators on device)
+ * @param red red color value
+ * @param green green color value
+ * @param blue blue color value
+ * @param flash flash rate, in Hz
+ * @return true if CAN message was successfully broadcasted
+ */
+bool shiftx_set_alert(uint8_t alert_id, uint8_t red, uint8_t green, uint8_t blue, uint8_t flash);
```

Comment:
I thought, consider passing a struct with the RGB and flash values instead of declaring parameters for each one.  Will make it easier to change many methods if you need to add anything in the future and plus it's a bit cleaner.

Reply:
Thanks for the suggestion. Will implement.